### DISCUSSION TITLE
Missing data

### DISCUSSION THREAD

**Discussion by PerpSearch** [benjaminjonard/koillection#1124] [Q&A] [answered]
I'm not sure you'll be able to replicate it because I've tried and haven't been able to.  
I've been adding items and editing items and several times a week I check to see if everything is still there.  It has been pretty sable since the last release anyway.  I was just checking now and noticed an item that lost all data in 14 fields, but not the primary image nor the tags I had attached.  The missing data fields were populated by choice list selections, a couple were just text entries, one date, one price, one rating, and one check box.  
I checked the history and it shows "the item was created 2 weeks ago" and the container's been up since 05/15 with 1.5.9 running.  I can easily replace the data as I back up the container twice a day, but I assumed you'd want to track it.

(edit) This was the only odd (and last) entry in the log file 
[15-May-2024 16:39:57] NOTICE: PHP message: PHP Warning:  Can't preload unlinked class Zenstruck\Foundry\Factory@anonymous: Unknown parent Zenstruck\Foundry\Factory in /var/www/koillection/vendor/zenstruck/foundry/src/functions.php on line 45

That said, the data was there yesterday.

**Comment by benjaminjonard** [comment]
Hello, it's weird because data fields other than select list are absolutely independent, nothing should be messing with them besides editing the related item. 
If you have some time (or next time it happens), I'd like to check if the data is still there in the database and i'd like to see what the API responds when asked to return an item's data.

First off all, connect to the API by executing the following command (just replace "YOUR_KOILLECTION_INSTANCE_URL", "YOUR_USERNAME" and "YOUR_PASSWORD"
```
curl --request POST \
  --url https://YOUR_KOILLECTION_INSTANCE_URL/api/authentication_token \
  --header 'Content-Type: application/json' \
  --header 'accept: application/json' \
  --data '{
    "username": "YOUR_USERNAME",
    "password": "YOUR_PASSWORD"
}'
```

You should get a token in the answer (something like "eyJ0eXAiOiJKV1Q...mDTPA")

Then query the item's data, copy paste the previous token in place of "YOUR_TOKEN" and put your item id in place of "YOUR_ITEM_ID"
```
curl --request GET \
  --url https://YOUR_KOILLECTION_INSTANCE_URL/api/items/YOUR_ITEM_ID/data \
  --header 'Authorization: Bearer YOUR_TOKEN' \
  --header 'Content-Type: application/json' \
  --header 'accept: application/json' \
  --data '{
}'
```

If you can please copy paste the result here

Thanks !

**Reply by PerpSearch** [reply]
OK, That looks doable, but I've already re entered what was missing so it will have to wait until next time (whenever that is) then I can follow up to assist.  I have no doubt it could be something on my end, but I can't imagine what.  It just seems so arbitrary. Anyway, thanks and it if pops up again I try to do what you requested.

C

**Reply by PerpSearch** [reply]
When I said that looks doable, you have to understand I can follow directions, but I'm fairly limited in my understanding here.  

I have Portainer and did the installation following a tutorial.  I'm looking at your instructions and presume when you say connect to the API and enter the following command you're asking me to open up what Portainer refers to as Container console where I can access kind of a command line access to the container which results in "Exec into container as default user using command bash" and entering curl --request GET \  I get an error, command not found.  

I take this to mean you need me to do this somewhere else or some other way.  I'm up for it, but need more guidance.  Thanks for your patience.

**Reply by benjaminjonard** [reply]
I'm not really familiar with Portainer unfortunately. But I meant you can run the commands on your own computer with a terminal (are you on Linux or Windows ?) 


The API is a bunch of urls (endpoints) that you can use to interact with Koillection without using the web interface
You can try to go on https://YOUR_KOILLECTION_URL/api, there you should see all the API endpoints. You should be able to replicate the 2 commands there.

First find the authentication endpoint `/api/authentication_token`, click on it to expand it, and click on the "Try it out" button. 
![Screenshot 2024-05-21 at 21-31-04 Koillection API - API Platform](https://github.com/benjaminjonard/koillection/assets/20560781/8436f614-47e6-4f37-9d98-a433bc1b1a21)

Fill in the username and password properties and click on "Execute"
![Screenshot 2024-05-21 at 21-38-32 Koillection API - API Platform](https://github.com/benjaminjonard/koillection/assets/20560781/6dadc1d0-d358-43a1-a899-3bebf8655288)

You should get an authentication token from it, copy it
![Screenshot 2024-05-21 at 21-39-01 Koillection API - API Platform](https://github.com/benjaminjonard/koillection/assets/20560781/e664eca6-9b52-4f60-859d-214443e61075)

At the very top of the page, click on the "Authorize" button 
![Screenshot 2024-05-21 at 21-39-17 Koillection API - API Platform](https://github.com/benjaminjonard/koillection/assets/20560781/df244294-f3f9-468e-81c5-7b01a5472d75)

and paste your token in it, click on authorize
![Screenshot 2024-05-21 at 21-39-35 Koillection API - API Platform](https://github.com/benjaminjonard/koillection/assets/20560781/784f1479-cbf8-485f-bb87-6e8875fde903)

You are now logged into the API, we can now get an item data fields.  
Find the endpoint `/api/items/{id}/data`, expand it and click on "Try it out" just like before
Fill in the item ID (ignore the page parameter) and execute. 
![Screenshot 2024-05-21 at 21-40-57 Koillection API - API Platform](https://github.com/benjaminjonard/koillection/assets/20560781/0487b54d-3213-47d8-82f3-c254a86c8a52)

You should get all the data linked to that item
![Screenshot 2024-05-21 at 21-41-07 Koillection API - API Platform](https://github.com/benjaminjonard/koillection/assets/20560781/a2dada3f-a507-4b8f-98b6-3ec2e2d9bb50)

This is a quick guide, I hope it's clear enough.
And no worries, thanks for your time

**Reply by PerpSearch** [reply]
I am amazed I got as far as I did, but I ran into this so I presume I did something wrong

![2024-05-21_225048](https://github.com/benjaminjonard/koillection/assets/92053831/8d725d43-121b-4600-9533-5fe6b797bfc4)

**Reply by PerpSearch** [reply]
. . . so assuming the Datum Identifier is the hash value or whatever I used it and got this

![2024-05-21_230514](https://github.com/benjaminjonard/koillection/assets/92053831/65dcc102-681b-4499-bc13-de79fa2ff613)

Which doesn't look right.  Pretty interesting though.

**Reply by benjaminjonard** [reply]
You are close but you use the wrong endpoint. I think you are using `/api/collections/{id}/data` instead of `/api/items/{id}/data`

The id you should use is the item ID, you can find it in the URL when looking at the item page.
(something like that  https://localhost/items/cac1ea23-a29b-461e-8f9c-b9b130d7a3e0 the ID would be cac1ea23-a29b-461e-8f9c-b9b130d7a3e0)

**Reply by PerpSearch** [reply]
Well, I just came up with 2 items missing data.  I was able to follow your tutorial (I think) and have copied the response.  This is what it looks like on my screen: 
![2024-05-28_233537](https://github.com/benjaminjonard/koillection/assets/92053831/44948779-2e70-4e65-b66d-144960fc9327)
and the part you're looking for is attached.
[response_1716954301695.json](https://github.com/benjaminjonard/koillection/files/15478488/response_1716954301695.json)

Hope this helps and thanks for looking into it and let me know if there's anything else you need.  I'm going to leave things they way they until you're done poking around.
Thanks again.

Charley

**Reply by benjaminjonard** [reply]
I won't be home for the next couple of days, I'll have a closer look when I'm back.
I just noticed that most of the data fields seem to have been updated at this date 2024-05-28 19:52:42. Do you remember doing anything on Koillection around that time ?

**Comment by PerpSearch** [comment]
I was working with it quite a bit yesterday, but I wasn't editing the items that were affected. I added a few things and edited a couple of other items, but the two that were affected I hadn't touched. I'm in no rush.  I'm just happy I was able to follow your directions to report back.  Chat later

**Comment by azuravian** [comment]
I'm a new user and this has been happening to me in real time.  I have a collection with a template attached.  I'm filling in entries.  Every so often when I add one, after clicking submit and returning to the list view, a different one will have all of its data removed.  The image below shows an example.  I had added Majisto's Magical Workshop with all fields complete.  I then added the Harry Potter & Cho Chang set and Majisto's row was as seen below.

![image](https://github.com/benjaminjonard/koillection/assets/7379812/b5ab7b03-c221-4eca-983d-fe5795c58fd9)

**_Edit:_** Quick update, it just happened again.  I added the Star Wars Diorama entry and immediately the Kingsley Shacklebolt entry (that was complete in the first image) is now clear of all data except name and images.  Although not shown, there is a secondary image that is part of the template.  That image also stays, but all other fields that are part of the template disappear.

![image](https://github.com/benjaminjonard/koillection/assets/7379812/081f6558-f847-4d96-895c-dc9e4e4dfd85)

**Reply by PerpSearch** [reply]
Interesting. Sounds like the problem I had a couple times, but I've been unable to replicate it.

**Reply by benjaminjonard** [reply]
Hello, I may have found the issue. Did you use the "Load common fields" feature ?

**Reply by azuravian** [reply]
I did not.  I was utilizing Power Automate, but it is just operating as a scraper and then clicking the new item button and entering data in.  Basically performing the same options I would do by hand.  After it clicks the submit button and goes back to the collection screen, I'll immediately see the data missing.  It also seems to be completely random as far as which entry is effected.  For example, it's not necessarily the most recent previous entry.

**Comment by azuravian** [comment]
OK, I was able to run the commands you specified earlier.  I'll post the returned json below.  It looks like each field shows both the date it was created (yesterday) and has an updated date (a few minutes ago).  The new entry has a creation date of very close to the updated date of the corrupted one, but they are about 9 seconds apart.  Interestingly, 9 seconds is about how long the new entry page is open to fill in everything.

[corrupted entry.json](https://github.com/user-attachments/files/15598188/corrupted.entry.json)
[new entry.json](https://github.com/user-attachments/files/15598187/new.entry.json)

**Comment by benjaminjonard** [comment, marked as answer]
The last release includes a fix for disappearing data fields. It would affect a random item within a collection when adding a new item, producing the same result as described here. Hopefully it will fix this issue
